java/ql/lib/semmle/code/java/security/PathSanitizer.qll

```diff
@@ -357,6 +357,21 @@ private predicate maybeNull(Expr expr) {
   )
 }
 
+/** A taint-tracking configuration for reasoning about tainted arguments. */
+private module TaintedArgConfig implements DataFlow::ConfigSig {
+  predicate isSource(DataFlow::Node src) {
+    src instanceof ActiveThreatModelSource or
+    src instanceof ApiSourceNode or
+    // for InlineFlowTest
+    src.asExpr().(MethodCall).getMethod().getName() = "source"
+  }
+
+  predicate isSink(DataFlow::Node sink) { exists(Call call | sink.asExpr() = call.getAnArgument()) }
```

Comment:
I worry this could be a performance problem - finding all global flow to arguments of calls. Better to make it more specific - something like `any(ConstructorCall constrCall | constrCall.getConstructedType() instanceof TypeFile).getArgument(0)`.

Reply:
Yes, I'm also concerned about that, which is why I'm currently running DCA.
Based on previous experimentation with this, I don't think restricting to `ConstructorCall` will necessarily be any better, but I'll give it a try if my current DCA experiment shows bad performance.

Do we have any better/pre-built way of checking whether an expression is tainted that doesn't require making a `TaintTracking` config?

Reply:
Yep, performance is bad. Running DCA on the `ConstructorCall` version to see if there's a difference.

Reply:
The `ConstructorCall` version still has bad performance (although not as bad as the `Call` version).